cmd/cli/dashboard.go

```diff
@@ -108,67 +106,43 @@ func (d *dashboardCmd) run() error {
 	// Select pod/s given the service data available
 	set := labels.Set(svc.Spec.Selector)
 	listOptions := metav1.ListOptions{LabelSelector: set.AsSelector().String()}
-	pods, err := v1ClientSet.Pods(settings.Namespace()).
-		List(context.TODO(), listOptions)
+	pods, _ := v1ClientSet.Pods(settings.Namespace()).List(context.TODO(), listOptions)
 
 	// Will select first running Pod available
-	it := 0
-	for {
-		if pods.Items[it].Status.Phase == "Running" {
+	var grafanaPod *corev1.Pod
+	for _, pod := range pods.Items {
+		if pod.Status.Phase == "Running" {
+			grafanaPod = &pod
 			break
 		}
-
-		it++
-		if it == len(pods.Items) {
-			return errors.Errorf("No running Grafana pod available.")
-		}
+	}
+	if grafanaPod == nil {
+		return errors.Errorf("No running Grafana pod available")
 	}
 
-	// Build http spdy-upgraded handler
-	dialer := createDialer(conf, v1ClientSet, pods.Items[it].GetName())
-
-	// StopChan is used to understand the lifecycle of the forwarding blocking op
-	// ReadyChan signals when the connection has been established and forwarding is in place
-	stopChan := make(chan struct{}, 1)
-	readyChan := make(chan struct{}, 1)
-	out, errOut := new(bytes.Buffer), new(bytes.Buffer)
-
-	forwardStr := fmt.Sprintf("%d:%d", d.localPort, d.remotePort)
-	fmt.Fprintf(d.out, "[+] Using forwarding: %s\n", forwardStr)
-
-	forwarder, err := portforward.New(dialer,
-		[]string{forwardStr},
-		stopChan,
-		readyChan,
-		out,
-		errOut)
+	portForwarder, err := NewPortForwarder(conf, clientSet, grafanaPod.Name, grafanaPod.Namespace, d.localPort, d.remotePort)
 	if err != nil {
-		return errors.Errorf("Failed to create forwarder: %s\n", err)
+		return errors.Errorf("Error setting up port forwarding: %s", err)
 	}
 
-	// Binding SIGINT & SIGTERM to trigger the closing routine
-	signal.Notify(d.sigintChan, syscall.SIGINT, syscall.SIGTERM)
-	go func() {
-		<-d.sigintChan // Blocking
-		fmt.Fprintf(d.out, "[+] SIGINT/TERM received, closing")
-		close(stopChan)
-	}()
-
-	// This routine blocks on readyChan till forwarding is setup and ready.
-	go func() {
-		<-readyChan
-
-		fmt.Fprintf(d.out, "[+] Port forwarding successful (localhost:%d)\n", d.localPort)
```

Comment:
Is there any output like this when the port forward is successful? I could see that being helpful just to communicate that things are working fine and nothing is hung up.

Reply:
I don't think it is necessary because in case of an error with port forwarding the output will be relayed back unlike before.

Reply:
I agree that it's not necessary, but my first thought whenever I run a command that produces no output is "what did I do wrong to make this hang," so it's just a usability thing. At least in the case that the browser isn't being opened.

Reply:
You should still see:
`[+] Starting Dashboard forwarding` regardless.